CalibTracker/SiStripESProducers/plugins/real/SiStripDelayESProducer.cc

```diff
@@ -16,53 +16,80 @@
 //
 //
 
+// system include files
+#include <memory>
 
+// user include files
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
 
-#include "CalibTracker/SiStripESProducers/plugins/real/SiStripDelayESProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
+#include "DataFormats/SiStripCommon/interface/SiStripConstants.h"
+#include "CalibFormats/SiStripObjects/interface/SiStripDelay.h"
+#include "CalibTracker/Records/interface/SiStripDependentRecords.h"
 
-
-SiStripDelayESProducer::SiStripDelayESProducer(const edm::ParameterSet& iConfig):
-  pset_(iConfig),
-  toGet(iConfig.getParameter<Parameters>("ListOfRecordToMerge"))
-{  
-  setWhatProduced(this);
+class SiStripDelayESProducer : public edm::ESProducer {
+ public:
+  SiStripDelayESProducer(const edm::ParameterSet&);
+  ~SiStripDelayESProducer() override{};
   
+  std::unique_ptr<SiStripDelay> produce(const SiStripDelayRcd&);
+   
+ private:
+  struct TokenSign {
+    TokenSign(edm::ESConsumesCollector& cc, const std::string& recordName, const std::string& label, int sumSign):
+      token_{cc.consumesFrom<SiStripBaseDelay, SiStripBaseDelayRcd>(edm::ESInputTag{"", label})},
+      recordName_{recordName},
+      label_{label},
+      sumSign_{sumSign}
+    {}
+    edm::ESGetToken<SiStripBaseDelay, SiStripBaseDelayRcd> token_;
+    std::string recordName_;
+    std::string label_;
+    int sumSign_;
+  };
+  std::vector<TokenSign> tokens_;
+};
+
+SiStripDelayESProducer::SiStripDelayESProducer(const edm::ParameterSet& iConfig) {
+  auto cc = setWhatProduced(this);
+
   edm::LogInfo("SiStripDelayESProducer") << "ctor" << std::endl;
 
+  for(const auto& pset: iConfig.getParameter<std::vector<edm::ParameterSet>>("ListOfRecordToMerge")) {
+    auto recordName = pset.getParameter<std::string>("Record");
+    auto label = pset.getParameter<std::string>("Label");
+
+    edm::LogInfo("SiStripDelayESProducer") << "[SiStripDelayESProducer::ctor] Going to get data from record " << recordName << " with label " << label;
+
+    // Is the "recordName" parameter really useful?
+    if(recordName == "SiStripBaseDelayRcd") {
+      tokens_.emplace_back(cc, recordName, label, pset.getParameter<int>("SumSign"));
+    }
+    else {
+      // Would an exception make sense?
+      edm::LogError("SiStripDelayESProducer") << "[SiStripDelayESProducer::ctor] Skipping the requested data for unexisting record " << recordName << " with tag " << label << std::endl;
```

Comment:
Input is read only if `Record` has a specific value, otherwise an error is printed. I have two questions
* now that the configuration is parsed in constructor, would an exception make sense (to point out a configuration error)?
* is the `Record` parameter really useful anymore? (possible implications being beyond this PR though)